### ISSUE TITLE
Missing scheduler.state_dict() when `Trainer.save_model`

### ISSUE THREAD

**Issue by MaximeSongIdris** [deepinv/deepinv#390]
When saving training to resume a training later on, if we are using a scheduler, we need to save its state_dict :

```
        torch.save({
            'epoch': epoch,
            'model_state_dict': model.state_dict(),
            'optimizer_state_dict': optimizer.state_dict(),
            'scheduler_state_dict': scheduler.state_dict()
        }, 'checkpoint.pth')
```

**Comment by MaximeSongIdris** [comment]
When loading from checkpoint:
https://github.com/deepinv/deepinv/blob/main/deepinv/training/trainer.py#L270

we need to pay attention to torch.load(), because some training may happen on a device "cuda:1" and not available when resuming training -> to make things proper, we can load everything on cpu first (instanciating model on cpu, loading checkpoint on cpu, loading weight on cpu, then moving model to gpu

**Comment by MaximeSongIdris** [comment]
https://deepinv.github.io/deepinv/_modules/deepinv/training/trainer.html#Trainer
When loading from ckpt_pretrained, `epoch_start = checkpoint["epoch"]` in `load_model` is wrong, as the model is already trained at this epoch.
It should be: `epoch_start = checkpoint["epoch"] + 1`

**Comment by Andrewwango** [comment]
For anyone working on this issue, please note the latest modifications in `load_model` in #386 #394

**Comment by Andrewwango** [comment]
Closed by #437
